From 61544a5249f0aadb9602d19b6b3cbe012170f465 Mon Sep 17 00:00:00 2001
From: Li Nan <linan122@huawei.com>
Date: Thu, 3 Nov 2022 22:31:34 +0800
Subject: [PATCH 266/595] [Huawei] blk-mq: fix kabi broken in struct bio

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187921

--------------------------------

Enable CONFIG_BLK_CGROUP_IOCOST will cause kabi broken, use reserved
fields to fix it.

Signed-off-by: Li Nan <linan122@huawei.com>
---
 include/linux/blk_types.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index 3e616267425e..c4e17f180669 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -238,9 +238,6 @@ struct bio {
 	 */
 	struct blkcg_gq		*bi_blkg;
 	struct bio_issue	bi_issue;
-#ifdef CONFIG_BLK_CGROUP_IOCOST
-	u64			bi_iocost_cost;
-#endif
 #endif
 
 #ifdef CONFIG_BLK_INLINE_ENCRYPTION
@@ -272,7 +269,11 @@ struct bio {
 	 * scsi log enhance feature
 	 */
 	KABI_USE(1, u64 bi_start_time_ns)
+#ifdef CONFIG_BLK_CGROUP_IOCOST
+	KABI_USE(2, u64 bi_iocost_cost)
+#else
 	KABI_RESERVE(2)
+#endif
 	KABI_RESERVE(3)
 	KABI_RESERVE(4)
 
-- 
2.17.1

